EIPS/eip-7496.md

```diff
@@ -0,0 +1,216 @@
+
+The `dataType` object MAY have a `valueMappings` object defined. If the `valueMappings` object is defined, the `valueMappings` object MUST be a mapping of `bytes32` values to numeric or unset `null` values.
+
+#### `boolean` Metadata Type
+
+The `boolean` metadata type allows for a boolean value to be set for a trait.
+
+The `dataType` object MAY have a `valueMappings` object defined. If the `valueMappings` object is defined, the `valueMappings` object MUST be a mapping of `bytes32` values to `boolean` or unset `null` values. The `boolean` values MUST be unique.
+
+If `valueMappings` is not used, the default trait values for `boolean` should be `bytes32(0)` for `false` and `bytes32(uint256(1))` (`0x0000000000000000000000000000000000000000000000000000000000000001`) for `true`.
+
+#### `epochSeconds` Metadata Type
+
+The `epochSeconds` metadata type allows for a numeric value to be set for a trait in seconds since the Unix epoch.
+
+The `dataType` object MAY have a `valueMappings` object defined. If the `valueMappings` object is defined, the `valueMappings` object MUST be a mapping of `bytes32` values to integer or unset `null` values.
+
+### Events
+
+Updating traits MUST emit one of:
+
+- `TraitUpdated`
+- `TraitUpdatedRange`
+- `TraitUpdatedRangeUniformValue`
+- `TraitUpdatedList`
+- `TraitUpdatedListUniformValue`
+
+For the `Range` events, the `fromTokenId` and `toTokenId` MUST be a consecutive range of tokens IDs and MUST be treated as an inclusive range.
+
+For the `List` events, the `tokenIds` MAY be in any order.
+
+It is RECOMMENDED to use the `UniformValue` events when the trait value is uniform across all token ids, so offchain indexers can more quickly process bulk updates rather than fetching each trait value individually.
+
+Updating the trait metadata URI MUST emit the event `TraitMetadataURIUpdated` so offchain indexers can be notified to query the contract for the latest changes via `getTraitMetadataURI()`.
+
+### `setTrait`
+
+If a trait defines `tokenOwnerCanUpdateValue` as `true`, then the trait value MUST be updatable by the token owner by calling `setTrait`.
+
+If the value the token owner is attempting to set is not valid, the transaction MUST revert. If the value is valid, the trait value MUST be updated and one of the `TraitUpdated` events MUST be emitted.
+
+If the trait has a `valueMappings` entry defined for the desired value being set, `setTrait` MUST be called with the corresponding `traitValue`.
+
+## Rationale
+
+Onchain traits can be used by contracts to get and mutate traits in a variety of different scenarios. For example, a contract that wants to entitle a token to a consumable benefit (e.g. a redeemable) can robustly reflect that onchain. Marketplaces can allow bidding on these tokens based on the trait value without having to rely on offchain state and exposing users to frontrunning attacks.
```

Comment:
This belongs in the Motivation section. The Rationale section should explain technical choices made within the proposal itself, while the Motivation section should justify the proposal as a whole.

The Rationale section could, for example, explain the reasoning behind the `validateOnSale` functionality (I assume it exists to invalidate off-chain offer messages?), or why `valueMappings` exists.

Reply:
thanks, updated